worlds/khrecom/Client.py

```diff
@@ -0,0 +1,226 @@
+        """Import kivy UI system and start running it as self.ui_task."""
+        from kvui import GameManager
+
+        class KHRECOMManager(GameManager):
+            logging_pairs = [
+                ("Client", "Archipelago")
+            ]
+            base_title = "Archipelago KHRECOM Client"
+
+        self.ui = KHRECOMManager(self)
+        self.ui_task = asyncio.create_task(self.ui.async_run(), name="UI")
+
+
+async def game_watcher(ctx: KHRECOMContext):
+    from .Locations import lookup_id_to_name
+    while not ctx.exit_event.is_set():
+        global death_link
+        if death_link and "DeathLink" not in ctx.tags:
+            await ctx.update_death_link(death_link)
+        if not death_link and "DeathLink" in ctx.tags:
+            await ctx.update_death_link(death_link)
+        if ctx.syncing == True:
+            sync_msg = [{'cmd': 'Sync'}]
+            if ctx.locations_checked:
+                sync_msg.append({"cmd": "LocationChecks", "locations": list(ctx.locations_checked)})
+            await ctx.send_msgs(sync_msg)
+            ctx.syncing = False
+        sending = []
+        victory = False
+        for root, dirs, files in os.walk(ctx.game_communication_path):
+            for file in files:
+                if file.find("send") > -1:
+                    st = file.split("send", -1)[1]
+                    if st != "nil":
+                        sending = sending+[(int(st))]
+                if file.find("victory") > -1:
+                    victory = True
+                if file.find("dlsend") > -1 and "DeathLink" in ctx.tags:
+                    st = file.split("dlsend", -1)[1]
+                    if st != "nil":
+                        if timegm(time.strptime(st, '%Y%m%d%H%M%S')) > ctx.last_death_link and int(time.time()) % int(timegm(time.strptime(st, '%Y%m%d%H%M%S'))) < 10:
+                            await ctx.send_death(death_text = "Sora was defeated!")
+        ctx.locations_checked = sending
+        message = [{"cmd": 'LocationChecks', "locations": sending}]
+        await ctx.send_msgs(message)
+        if not ctx.finished_game and victory:
+            await ctx.send_msgs([{"cmd": "StatusUpdate", "status": ClientStatus.CLIENT_GOAL}])
+            ctx.finished_game = True
+        await asyncio.sleep(0.1)
+
+
+def launch():
+    async def main(args):
+        ctx = KHRECOMContext(args.connect, args.password)
+        ctx.server_task = asyncio.create_task(server_loop(ctx), name="server loop")
+        if gui_enabled:
+            ctx.run_gui()
+        ctx.run_cli()
+        progression_watcher = asyncio.create_task(
+            game_watcher(ctx), name="KHRECOMProgressionWatcher")
+
+        await ctx.exit_event.wait()
+        ctx.server_address = None
+
+        await progression_watcher
+
+        await ctx.shutdown()
+
+    import colorama
+
+    parser = get_base_parser(description="KHRECOM Client, for text interfacing.")
+
+    args, rest = parser.parse_known_args()
+    colorama.init()
```

Comment:
`init` is being deprecated in favor of `just_fix_windows_console.` I didn't pick the name.